core/src/io/anuke/mindustry/world/blocks/power/PowerGraph.java

```diff
@@ -107,11 +161,11 @@ public void add(Tile tile){
         tile.entity.power.graph = this;
         all.add(tile);
 
-        if(tile.block().outputsPower){
+        if (tile.block().outputsPower && tile.block().consumesPower) {
+            batteries.add(tile);
+        } else if (tile.block().outputsPower) {
             producers.add(tile);
-        }
-
-        if(tile.block().consumesPower){
+        } else if (tile.block().consumesPower) {
```

Comment:
Please fix your formatter; statements should have no extra spaces, e.g. `} else if (` -> `}else if(`.
This occurs throughout the PR.